tests/rados/test_bluestore_checksum.py

```diff
@@ -0,0 +1,71 @@
+""" Module to verify scenarios related to BlueStore Checksum algorithms"""
+
+from ceph.ceph_admin import CephAdmin
+from ceph.rados.core_workflows import RadosOrchestrator
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    """
+    #CEPH-83571646
+    Apply all the applicable different checksum algorithms on pools backed by bluestore
+    Valid algos: none, crc32c, crc32c_16, crc32c_8, xxhash32, xxhash64
+    1. Create individual replicated pools for each checksum
+    2. Verify the default checksum algorithm is crc32c
+    3. Set different checksum algorithm as global and for each pool
+    4. Verify the checksum algo being set correctly
+    5. Write data to each pool using rados bench
+    6. cleanup - Remove all the pools created
+    """
+    log.info(run.__doc__)
+    config = kw["config"]
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+    rados_obj = RadosOrchestrator(node=cephadm)
+    pool_list = []
+    checksum_list = config.get("checksums")
+
+    log.info("Running test case to verify BlueStore checksum algorithms")
+
+    try:
+        # verify default checksum value
+        out, _ = cephadm.shell(["ceph config get osd bluestore_csum_type"])
+        log.info(f"BlueStore OSD default checksum: {out} | Expected: crc32c")
+        assert "crc32c" in out
+
+        for checksum in checksum_list:
+            # set the global checksum value
+            out, _ = cephadm.shell(
+                [f"ceph config set osd bluestore_csum_type {checksum}"]
+            )
+
+            # verify the newly set global checksum value
+            out, _ = cephadm.shell(["ceph config get osd bluestore_csum_type"])
+            assert checksum in out
+            log.info(f"global checksum set verified - {out}")
+
+            # create pool with given config
+            pool_name = f"re_pool_3_{checksum}"
+            rados_obj.create_pool(pool_name=pool_name)
```

Comment:
Can we create few replicated pool and few erasure and verify instead of all replicated?